synapse/http/servlet.py

```diff
@@ -125,14 +125,15 @@ def parse_string(request, name, default=None, required=False,
             is not one of those allowed values.
     """
     return parse_string_from_args(
-        request.args, name, default, required, allowed_values, param_type,
+        request.args, name.encode('ascii'), default, required, allowed_values,
```

Comment:
I feel like this might be the wrong place for this. parse_string is currently a synonym for parse_string_from_args, and you're changing that.

We should probably make it so that we *don't* have two identical functions, but meantime: how about putting this in parse_string_from_args instead?